docs/proposals/dynamic-openshift-settings-configmaps.md

```diff
@@ -0,0 +1,70 @@
+# Changing OpenShift settings dynamically through ConfigMaps
+
+## Problem
+
+Currently, there are settings contained withing the master-config.yaml file that
+are of value to an unpriviledged user. Specifically, the `autoProvisionEnabled`
+boolean under `jenkinsPipelineConfig` is only available to users with the
+ability to write to the master-config.
+
+Handling settings like this through project annotations was explored, but
+because unpriviledged users cannot annotate their own projects, it was
+abandoned.
+
+## Design proposal:
+
+- In each project there can exist a ConfigMap annotated with
+  'project.openshift.io/configuration'
+
+- The ConfigMap will contain the key/value pair of the setting to be overridden
+  (key) and the value to use.
+
+- Only certain settings would be covered under this method.
+
+- For values with more than one entry, e.g. `corsAllowedOrigins`, yaml or json
+  array notation could be used.
+
+- (Optional) Have a utility function for consolidating settings over multiple
+  ConfigMaps. When settings overlap, output a warning telling the user that
+  behavior may be nondeterministic
+
+- (Optional) In addition to project-specific modifications to cluster-wide
+  settings, we could also introduce project-wide settings that would be handled
+  here.
+
+## Issues:
+
+- We will likely not be exposing all settings in this way, so there needs to be
+  an easy way for affected users to discover these.
```

Comment:
it's worth making it clear that this is not going to automatically expose all the settings, ie that logic will need to be implemented for each setting for which we want to provide project-level config support.  but it doesn't have to be an "issue".